### PR TITLE
FEAT: allow additional artifacts and dry-runs for critical workflows

### LINKED ISSUE #138: Lack of consistency on paths for released artifacts
The ``release-to-PyPI`` and ``release-to-Github`` are not consistent.

The PyPI action download the artifacts (all) to a folder called ``${{ inputs.library-name}}-artifacts`` but then, the ``release-to-github`` action check the folder ``other-artifacts``.

### LINKED ISSUE #139: Implement ``dry-run`` option for critical actions
Our actions should have a ``dry-run`` option for the important/only-can-do-once things... like release to PyPI.

### LINKED ISSUE #140: Allow to publish additional artifacts in release-github action
So far, the action uploads everything. I don't think this is a good approach. Everything should be optional.

```yml

    - name: "Release to GitHub"
      uses: softprops/action-gh-release@v1
      with:
        files: |
          ${{ inputs.library-name }}/*.whl
          ${{ inputs.library-name }}/*.tar.gz
          documentation-html.zip
          documentation-pdf.zip
          other-artifacts/**/*-wheelhouse-*.zip
```

### PR DESCRIPTION
**Pull request by jorgepiloto**
Fix #138, fix #139 and fix #140 by allowing to download additional artifacts. Related to #178 too.

### CODE REVIEW

File: _release-pypi/action.yml [hunk 1]
```diff
@@ -43,6 +43,15 @@ inputs:
     required: true
     type: string
 
+  dry-run:
+    description: >
+      Whether to run or not this action in testing mode. Testing mode executes
+      all the action steps except the releasing artifacts one. This allows
+      developers to verify the health of the action once integrated in their
+      CI/CD pipelines without actually publishing artifacts.
+    required: true
```
Comment (MaxJPRey) [inline _release-pypi/action.yml:52]: I am wondering here if we should have the same arguments as for `release-github` action.
I was thinking that usually when you do a release, you want to publish on PyPI -so the logic should be similar-.
But it is actually not always true. So I am open to any decisions.

```suggestion
    required: false
    default: false
```

Comment (RobPasMue) [reply _release-pypi/action.yml:52]: Dry-run should definitely be optional. This is must be optional.

Comment (RobPasMue) [reply _release-pypi/action.yml:52]: Ahh now I see... This is the private action. In that case, that's fine

**Review by RobPasMue — APPROVED** [review]

File: _release-pypi/action.yml (hunk 1 above)
Comment (jorgepiloto) [reply _release-pypi/action.yml:52]: That's it, @RobPasMue. This is the private action. The convention is that private actions do not have optional arguments to be transparent when called from public actions.

By default, the "dry-run" is disabled in:

* `release-pypi-public`
* `release-pypi-private`
* `release-pypi-test`
* `release-github`

Comment (MaxJPRey) [reply _release-pypi/action.yml:52]: OK make sense. I did not notice that it was private. Thanks for the clarification.

**Review by MaxJPRey — APPROVED** [review]

**Pull request merged**